server/src/main/java/org/apache/iotdb/db/engine/storagegroup/timeindex/OldFileTimeIndex.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.engine.storagegroup.timeindex;
+
+import org.apache.iotdb.db.engine.StorageEngine;
+import org.apache.iotdb.db.exception.PartitionViolationException;
+import org.apache.iotdb.db.rescon.CachedStringPool;
+import org.apache.iotdb.db.utils.SerializeUtils;
+import org.apache.iotdb.tsfile.utils.FilePathUtils;
+import org.apache.iotdb.tsfile.utils.RamUsageEstimator;
+import org.apache.iotdb.tsfile.utils.ReadWriteIOUtils;
+
+import io.netty.util.internal.ConcurrentSet;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class OldFileTimeIndex implements ITimeIndex {
+
+  protected static final Map<String, String> cachedDevicePool =
+      CachedStringPool.getInstance().getCachedPool();
+
+  /** start time */
+  protected long startTime;
+
+  /** end times. The value is Long.MIN_VALUE if it's an unsealed sequence tsfile */
+  protected long endTime;
+
+  /** devices */
+  protected Set<String> devices;
+
+  public OldFileTimeIndex() {
+    this.devices = new ConcurrentSet<>();
+    this.startTime = Long.MAX_VALUE;
+    this.endTime = Long.MIN_VALUE;
+  }
+
+  public OldFileTimeIndex(Set<String> devices, long startTime, long endTime) {
+    this.startTime = startTime;
+    this.endTime = endTime;
+    this.devices = devices;
+  }
+
+  @Override
+  public void serialize(OutputStream outputStream) throws IOException {
+    ReadWriteIOUtils.write(devices.size(), outputStream);
+    Set<String> stringMemoryReducedSet = new ConcurrentSet<>();
+    for (String device : devices) {
+      // To reduce the String number in memory,
+      // use the deviceId from cached pool
+      stringMemoryReducedSet.add(cachedDevicePool.computeIfAbsent(device, k -> k));
+      ReadWriteIOUtils.write(device, outputStream);
+    }
+    ReadWriteIOUtils.write(startTime, outputStream);
+    ReadWriteIOUtils.write(endTime, outputStream);
+    devices = stringMemoryReducedSet;
+  }
+
+  @Override
+  public OldFileTimeIndex deserialize(InputStream inputStream) throws IOException {
```

Comment:
May be let the deserialize return the latest FileTimeIndex is much better. And leave other interfaces empty.

Reply:
Great idea! I let other methods throw UnsupportedOperationException instead.